waiter/src/waiter/auth/authentication.clj

```diff
@@ -22,20 +22,6 @@
 
 (def ^:const AUTH-COOKIE-NAME "x-waiter-auth")
 
-(defprotocol Authenticator
```

Comment:
I like holding on to this protocol because it clearly specifies the signature of the function that must be supported. Maybe I am biased by years of using Java interfaces. Same comment for the `authorizer` defined in the scheduler below. It is also good for consistency, since most of our `create-component` calls return instances of protocols.